flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestingJobResultStore.java

```diff
@@ -80,33 +82,65 @@ private TestingJobResultStore(
     }
 
     @Override
-    public void createDirtyResult(JobResultEntry jobResultEntry) throws IOException {
-        createDirtyResultConsumer.accept(jobResultEntry);
+    public CompletableFuture<Void> createDirtyResultAsync(JobResultEntry jobResultEntry) {
+        try {
+            createDirtyResultConsumer.accept(jobResultEntry);
+        } catch (IOException e) {
+            CompletableFuture<Void> future = new CompletableFuture<>();
+            future.completeExceptionally(e);
+            return future;
+        }
+        return CompletableFuture.completedFuture(null);
     }
 
     @Override
-    public void markResultAsClean(JobID jobId) throws IOException {
-        markResultAsCleanConsumer.accept(jobId);
+    public CompletableFuture<Void> markResultAsCleanAsync(JobID jobId) {
+        try {
+            markResultAsCleanConsumer.accept(jobId);
+        } catch (IOException e) {
```

Comment:
You're not following the contract here, I guess. :thinking: Usually for this `Testing*` implementations, we would change the callback type (e.g. a `ThrowingConsumer<T>` becomes a `Function<T, CompletableFuture<Void>`). That enables us to have the entire logic being specified in the test method.

Reply:
I've fixed the builder of TestingJobResultStore.